Signal/src/ViewControllers/DebugUI/DebugUIMessages.m

```diff
@@ -3372,23 +3372,32 @@ + (void)sendRandomAttachment:(TSThread *)thread uti:(NSString *)uti length:(NSUI
                                completion:nil];
 }
 
-+ (OWSSignalServiceProtosEnvelope *)createEnvelopeForThread:(TSThread *)thread
++ (SSKEnvelope *)createEnvelopeForThread:(TSThread *)thread
 {
     OWSAssert(thread);
 
-    OWSSignalServiceProtosEnvelopeBuilder *builder = [OWSSignalServiceProtosEnvelopeBuilder new];
-
-    [builder setTimestamp:[NSDate ows_millisecondTimeStamp]];
+    uint64_t timestamp = [NSDate ows_millisecondTimeStamp];
+    NSString *source = ^{
+        if ([thread isKindOfClass:[TSGroupThread class]]) {
+            TSGroupThread *gThread = (TSGroupThread *)thread;
+            return gThread.groupModel.groupMemberIds[0];
+        } else if ([thread isKindOfClass:[TSContactThread class]]) {
+            TSContactThread *contactThread = (TSContactThread *)thread;
+            return contactThread.contactIdentifier;
```

Comment:
Should this be a method on `TSThread`?

Reply:
I don't think so - In general we can't assume the sender is `groupMemberIds[0]`,  it's just expedient debug code.